Upgrading to 0.7.1

I've been a old baremaps version.
I've upgraded it to 0.7.1
But i aml getting an error
```
[ERROR] 2023-06-16 09:58:40.934 [servicetalk-global-executor-1-2] HttpExceptionMapperServiceFilter - Unexpected exception during service processing for connection=[id: 0x845aa73d, L:/[0:0:0:0:0:0:0:1%0]:9000 - R:/[0:0:0:0:0:0:0:1%0]:45702], request='GET /tiles/14/8311/5658.mvt HTTP/1.1'. Trying to return: 500 Internal Server Error
java.lang.NullPointerException: Cannot invoke "java.nio.ByteBuffer.duplicate()" because the return value of "org.apache.baremaps.database.tile.TileStore.read(org.apache.baremaps.database.tile.Tile)" is null

```

Tile set is pretty strait foward
```
{
  "tilejson": "2.1.0",
  "tiles": ["http://localhost:9000/tiles/{z}/{x}/{y}.mvt"],
  "vector_layers": [
    {
      "id": "activite",
      "queries": [
        {
          "minzoom": 13,
          "maxzoom": 20,
          "sql": "SELECT uuid_hash(id)::integer, jsonb_build_object('nom', nom::text) as tags, geom FROM nexsis.activite"
        }
      ]
    }
  ]
}
```
And styleset also
```
{
  "version" : 8,
  "sources" : {
    "baremaps" : {
      "type" : "vector",
      "url" : "http://localhost:9000/tiles.json"
    }
  },
  "layers" : [
    {
      "id" : "activite",
      "type" : "circle",
      "source" : "baremaps",
      "source-layer" : "activite",
      "layout" : {
        "visibility" : "visible"
      }
    }
  ],
  "center" : [ 2.66, 48.5395 ],
  "metadata" : { "maputnik:renderer" : "mbgljs" },
  "zoom" : 14
}
```

Select in database is ok
```
   abs     |                 tags                  |                        geom                        
------------+---------------------------------------+----------------------------------------------------
 2045938576 | {"nom": "{activite_ambassade}"}       | 0101000020110F0000576E0B891F0C1241823906CE34A45741
 1279007087 | {"nom": "{activite_justice}"}         | 0101000020110F000030581D41860C12412B5516082CA45741
 2048781369 | {"nom": "{activite_poste}"}           | 0101000020110F0000C6699A4CDE0C1241A016660726A45741
  522579177 | {"nom": "{activité_voiture}"}         | 0101000020110F0000C718493C3B0D124173307A411DA45741
  979850116 | {"nom": "{activite_musee}"}           | 0101000020110F0000ADB4FFB67F0D1241B0A32EDE15A45741
 2116634689 | {"nom": "{activite_CS}"}              | 0101000020110F0000CA76A61BF50D1241851B47180DA45741
  596747879 | {"nom": "{activite_magasin}"}         | 0101000020110F0000B0125D96390E12419F872DDC03A45741
```



Any help would be appreciated

--- a/baremaps-core/src/main/java/org/apache/baremaps/tilestore/TileCache.java
+++ b/baremaps-core/src/main/java/org/apache/baremaps/tilestore/TileCache.java
@@ -51,19 +51,19 @@ public class TileCache implements TileStore {
   /** {@inheritDoc} */
   @Override
   public ByteBuffer read(TileCoord tileCoord) throws TileStoreException {
-    return cache.get(tileCoord, t -> {
+    var buffer = cache.get(tileCoord, t -> {
       try {
-        var buffer = tileStore.read(t);
-        if (buffer == null) {
-          return null;
-        } else {
-          return buffer;
-        }
+        return tileStore.read(t);
       } catch (TileStoreException e) {
         logger.error("Unable to read the tile.", e);
         return null;
       }
-    }).duplicate();
+    });
+    if (buffer == null) {
+      return null;
+    } else {
+      return buffer.duplicate();
+    }
   }
 
   /** {@inheritDoc} */

